packages/astro/src/core/routing/manifest/create.ts

```diff
@@ -505,28 +493,26 @@ export function createRouteManifest(
 	}
 
 	const injectedRoutes = createInjectedRoutes(params);
-	for (const [, routes] of Object.entries(injectedRoutes)) {
-		for (const route of routes) {
-			routeMap.set(route.route, route);
-		}
+	for (const route of injectedRoutes) {
+		routeMap.set(route.route, route);
 	}
 
 	const redirectRoutes = createRedirectRoutes(params, routeMap, logger);
 
 	const routes: RouteData[] = [
-		...injectedRoutes['legacy'].sort(routeComparator),
-		...[...fileBasedRoutes, ...injectedRoutes['normal'], ...redirectRoutes['normal']].sort(
+		...[
+			...fileBasedRoutes,
+			...injectedRoutes,
+			...redirectRoutes
+		].sort(
 			routeComparator,
 		),
-		...redirectRoutes['legacy'].sort(routeComparator),
 	];
 
 	// Report route collisions
-	if (config.experimental.globalRoutePriority) {
-		for (const [index, higherRoute] of routes.entries()) {
-			for (const lowerRoute of routes.slice(index + 1)) {
-				detectRouteCollision(higherRoute, lowerRoute, config, logger);
-			}
+	for (const [index, higherRoute] of routes.entries()) {
+		for (const lowerRoute of routes.slice(index + 1)) {
+			detectRouteCollision(higherRoute, lowerRoute, config, logger);
 		}
 	}
```

Comment:
As a note, this collision report is the warning part that is currently incorrect due to this prioritization happening before the `export const prerender` is read from the files.

The entire prioritization is affected, but this is the part that yells at the users, so it is more visible.